api/src/test/java/org/apache/iceberg/io/TestCloseableIterable.java

```diff
@@ -91,10 +90,9 @@ public void testConcatWithEmptyIterables() {
     // This will throw a NoSuchElementException
     CloseableIterable<Integer> concat5 =
         CloseableIterable.concat(Lists.newArrayList(empty, empty, empty));
-    AssertHelpers.assertThrows(
-        "should throw NoSuchElementException",
-        NoSuchElementException.class,
-        () -> Iterables.getLast(concat5));
+
+    Assertions.assertThatThrownBy(() -> Iterables.getLast(concat5))
+        .isInstanceOf(NoSuchElementException.class);
```

Comment:
would be good to check for the specific error msg here

Reply:
thanks for the review. I don't see any error message in this case. 
Refer - https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Iterables.java#L849

Reply:
ah ok, thanks for checking. Alternatively we could use `.hasMessage(null)` to indicate that there's no message I guess